src/dnaio/exceptions.py

```diff
@@ -43,3 +43,28 @@ class FastaFormatError(FileFormatError):
     """
 
     format = "FASTA"
+
+
+class IsNotAscii(ValueError):
+    """
+    Some part of a FASTA or FASTQ file is not ASCII encoded
+
+    Attributes:
+        field: Description of the field ("name", "sequence", "qualities" or similar)
+            in which non-ASCII characters were found
+        value: Unicode string that was intended to be assigned to the field
+    """
+
+    def __init__(self, field: str, value: str):
+        self.value = value
+        self.field = field
+
+    def __str__(self):
+        detail = ""
+        try:
+            self.value.encode("ascii")
+        except UnicodeEncodeError as e:
+            detail = (
+                f", but found '{self.value[e.start:e.end]}' at position {e.start + 1}"
```

Comment:
I don't know if the error is confusing to normal users, but it would be confusing to me (a slightly less than normal user). 
I would load the string and type: `my_string[position]` and that gives back the wrong character. Because this position is one-based. Can it be renamed to `at index {e.start}`?

Reply:
Hm, the idea is that the text can be printed unchanged when a parse error occurs. Cutadapt does this, for example:
```
cutadapt: error: Error in FASTA file at line 5: 'name' in sequence file must be ASCII encoded, but found '☺' at position 1
```
It’s also consistent with the 1-based line number in `FileFormatError.__str__`: https://github.com/marcelm/dnaio/blob/095c08cf1fc7957a9e739342a4aef03344cc874e/src/dnaio/exceptions.py#L28

I see what you mean, though. IMO, enduser-visible numbers should be 1-based, and developers should get 0-based ones, but the exception message is shown both to users and developers, so there’s a conflict. IDK, maybe directly showing `str()` representations of exceptions to the user isn’t the best architecture (but it is pragmatic).